contracts/multi-token/tests/utils.rs

```diff
@@ -259,24 +276,55 @@ pub fn transform_internal(
     }
     let res = mtk.send(
         from,
-        MyMTKAction::Transform {
+        MtkAction::Transform {
             id: token_id,
             amount,
             nfts,
         },
     );
 
-    let codec = MTKEvent::Transfer {
-        operator: from.into(),
+    let codec = Ok::<MtkEvent, MtkError>(MtkEvent::Transfer {
         from: ActorId::zero(),
         to: ActorId::zero(),
         ids,
         amounts: vec![NFT_COUNT; amount as usize],
-    }
+    })
     .encode();
     assert!(res.contains(&(from, codec)));
 }
 
+pub fn approve(mtk: &Program<'_>, from: u64, to: u64, error: Option<MtkError>) {
+    let res = mtk.send(from, MtkAction::Approve { account: to.into() });
+
+    let codec = Ok::<MtkEvent, MtkError>(MtkEvent::Approval {
+        from: from.into(),
+        to: to.into(),
+    })
+    .encode();
+
+    if let Some(error) = error {
+        assert!(res.contains(&(from, Err::<MtkEvent, MtkError>(error).encode())));
+    } else {
+        assert!(res.contains(&(from, codec)));
+    }
+}
+
+pub fn revoke_approval(mtk: &Program<'_>, from: u64, to: u64, error: Option<MtkError>) {
+    let res = mtk.send(from, MtkAction::RevokeApproval { account: to.into() });
+
+    let codec = Ok::<MtkEvent, MtkError>(MtkEvent::RevokeApproval {
+        from: from.into(),
+        to: to.into(),
+    })
+    .encode();
+
+    if let Some(error) = error {
+        assert!(res.contains(&(from, Err::<MtkEvent, MtkError>(error).encode())));
+    } else {
+        assert!(res.contains(&(from, codec)));
+    }
```

Comment:
I'd rewrite a little bit.

```suggestion
    let expected = if let Some(error) = error {
        Err(error)
    } else {
        Ok(MtkEvent::RevokeApproval {
            from: from.into(),
            to: to.into(),
        })
    };
    assert!(res.contains(&(from, expected.encode())));
```